src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnRpService.java

```diff
@@ -0,0 +1,282 @@
+package it.smartcommunitylab.aac.webauthn.service;
+
+import java.security.SecureRandom;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.yubico.webauthn.AssertionRequest;
+import com.yubico.webauthn.AssertionResult;
+import com.yubico.webauthn.FinishAssertionOptions;
+import com.yubico.webauthn.FinishRegistrationOptions;
+import com.yubico.webauthn.RegistrationResult;
+import com.yubico.webauthn.RelyingParty;
+import com.yubico.webauthn.StartAssertionOptions;
+import com.yubico.webauthn.StartRegistrationOptions;
+import com.yubico.webauthn.data.AuthenticatorAssertionResponse;
+import com.yubico.webauthn.data.AuthenticatorAttestationResponse;
+import com.yubico.webauthn.data.AuthenticatorSelectionCriteria;
+import com.yubico.webauthn.data.AuthenticatorTransport;
+import com.yubico.webauthn.data.ByteArray;
+import com.yubico.webauthn.data.ClientAssertionExtensionOutputs;
+import com.yubico.webauthn.data.ClientRegistrationExtensionOutputs;
+import com.yubico.webauthn.data.PublicKeyCredential;
+import com.yubico.webauthn.data.PublicKeyCredentialCreationOptions;
+import com.yubico.webauthn.data.ResidentKeyRequirement;
+import com.yubico.webauthn.data.UserIdentity;
+import com.yubico.webauthn.data.UserVerificationRequirement;
+
+import org.springframework.util.StringUtils;
+
+import it.smartcommunitylab.aac.SystemKeys;
+import it.smartcommunitylab.aac.core.service.SubjectService;
+import it.smartcommunitylab.aac.model.Subject;
+import it.smartcommunitylab.aac.webauthn.auth.WebAuthnAuthenticationException;
+import it.smartcommunitylab.aac.webauthn.model.WebAuthnCredentialCreationInfo;
+import it.smartcommunitylab.aac.webauthn.model.WebAuthnLoginResponse;
+import it.smartcommunitylab.aac.webauthn.model.WebAuthnRegistrationResponse;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnCredential;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnCredentialsRepository;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnUserAccount;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnUserAccountRepository;
+
+public class WebAuthnRpService {
+
+    WebAuthnUserAccountRepository webAuthnUserAccountRepository;
+
+    WebAuthnCredentialsRepository webAuthnCredentialsRepository;
+
+    SubjectService subjectService;
```

Comment:
As of now I do not think I can remove it because of the `@NotNull` constraint on subject_id in the account [entity](https://github.com/civts/AAC/blob/411eed3030ebc6a11fefd4ac3a0a7e64575d7c69/src/main/java/it/smartcommunitylab/aac/webauthn/persistence/WebAuthnUserAccount.java#L27)